snapcraft/internal/yaml.py

```diff
@@ -144,14 +143,19 @@ def _process_parts(self):
             plugin_name = properties.pop('plugin', None)
             if not plugin_name:
                 logger.info(
-                    'Searching the wiki to compose part "{}"'.format(
-                        part_name))
-                with contextlib.suppress(KeyError):
-                    properties = self._wiki.compose(part_name, properties)
+                    'Searching in the remote parts cache for part '
+                    '{!r}'.format(part_name))
+                try:
+                    properties = self._remote_parts.compose(
+                        part_name, properties)
                     plugin_name = properties.pop('plugin', None)
-                    # The wiki still supports using 'type' for snapcraft 1.x
-                    if 'type' in properties:
-                        del properties['type']
+                except KeyError:
+                    logger.error(
+                        'A part without a plugin defined will be searched for '
+                        'in the defined remote parts, {!r} is missing the '
+                        '`plugin` entry and is not defined in the current '
+                        'remote parts cache, try to run `snapcraft update` '
+                        'to refresh'.format(part_name))
```

Comment:
This is hard to read, three commas in a sentence. I would remove the first sentence and just say: {!r} is not defined in the current remote parts cache, try to run `snapcraft update` to refresh. And maybe add a help entry were we explain more? So we can finish the message with: Type snapcraft help remote-parts for more information.
Just my opinion, I'm not sure my proposal is better.